feat: Database Monitoring MSSQL documentation [Feb Release] (#20030)
newrelic/docs-website: src/content/docs/infrastructure/host-integrations/host-integrations-list/microsoft-sql/microsoft-sql-server-query-performance.mdx

---
title: Microsoft SQL Server's integration query performance monitoring
tags:
  - Integrations
  - On-host integrations
  - On-host integrations list
  - Configuration
metaDescription: Configuration New Relic's Kafka integration.
freshnessValidatedDate: never
---

<img
  title="Query Performance monitoring"
  alt="Query Performance monitoring"
  src="/images/mssql-query-performance.png"
/>

You can view query-level analysis to monitor your SQL queries within your MSSQL database to assess their performance and impact. This feature provides insights into execution times, resource consumption, and potential bottlenecks, enabling you to optimize database operations. For more information

## Prerequisites:

> * Microsoft SQL Server supports 2017, 2019, and 2022.

Reviewer: ```suggestion
* Microsoft SQL Server 2017, 2019, or 2022.
```

* DB user should be a member of the sysadmin fixed server role or have the VIEW SERVER STATE permission

> To Enable the Query performance monitoring feature, follow these steps:

Reviewer: ```suggestion
To enable the Query performance monitoring feature, follow these steps:
```

<Steps>
  <Step>
    ## To Enable SQL Server and Windows Authentication mode

>     Enable the login on the SQL server and Windows Authentication mode. For more information, refer Microsoft change authentication mode [documentation](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/change-server-authentication-mode?view=sql-server-ver16&tabs=ssms).

Reviewer: ```suggestion
    Enable the login on the SQL Server and Windows Authentication mode. For more information, refer Microsoft change authentication mode [documentation](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/change-server-authentication-mode?view=sql-server-ver16&tabs=ssms).
```

  </Step>
  <Step>
    ## To Enable TCP
    1. Ensure TCP is enabled for remote access on your instance using the [documentation](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/configure-a-server-to-listen-on-a-specific-tcp-port?view=sql-server-ver16)

>     2. Check Firewall Settings:

Reviewer: ```suggestion
    2. Check firewall settings:
```

>     - Ensure that your firewall allows traffic on port 1433(this is the port on which MSSQL Server is listening).

Reviewer: ```suggestion
    - Ensure that your firewall allows traffic on port 1433 (this is the port on which MSSQL Server is listening).
```

    - Open Windows Defender Firewall with Advanced Security.

>     - Click on Inbound Rules on the left.

Reviewer: ```suggestion
    - In the left menu, click **Inbound Rules**.
```

    - Find any rule for SQL Server (typically named SQL Server (TCP-In) or SQL Server TCP 1433).
    - If a rule doesn't exist, create a new rule to allow TCP traffic on port 1433.
  </Step>
  <Step>
    ## To Enable the query store [#queryStore]
    Run the following command
    ```sql
    ALTER DATABASE DATABASE_NAME SET QUERY_STORE = ON ( QUERY_CAPTURE_MODE = ALL, DATA_FLUSH_INTERVAL_SECONDS = 900 )
    ```
  </Step>
  <Step>
    ## Run the below query to confirm if the query store is enabled [#confirmQueryStore]

>     if the output of the query is 1 then we can assume the query store is enabled

Reviewer: ```suggestion
    If the output of the query is 1 then we can assume the query store is enabled
```

      * QUERY_MONITORING_COUNT_THRESHOLD - The number of records for each query performance metrics - Defaults to 20

      ```yaml
        integrations:
        - name: nri-mssql
        env:
          HOSTNAME: localhost
          PORT: 1433
          USERNAME: USERNAME
          PASSWORD: PASSWORD
          ENABLE_SSL: true
          TRUST_SERVER_CERTIFICATE: true
          # Enable collection of detailed query performance metrics.
          ENABLE_QUERY_MONITORING: true
          # Threshold in milliseconds for query response time. If response time exceeds this threshold, the query will be considered slow.
          QUERY_MONITORING_RESPONSE_TIME_THRESHOLD: 500
          # Maximum number of queries returned in query analysis results.
          QUERY_MONITORING_COUNT_THRESHOLD: 20
          # Interval in seconds for fetching grouped slow queries; Should always be same as mysql-config interval.
          QUERY_MONITORING_FETCH_INTERVAL: 15
        interval: 15s
        labels:
          environment: production
        inventory_source: config/mssql
      ```
  </Step>
  <Step>
    ## Find your data

>     You can use our pre-built dashboard template to monitor your MSSQL metrics. Follow these steps to use our pre-built dashboard template:

Reviewer: ```suggestion
    You can use our prebuilt dashboard template to monitor your MSSQL metrics. Follow these steps to use our prebuilt dashboard template:
```

    1. From [one.newrelic.com](https://one.newrelic.com), go to the <DNT>**+ Integrations & Agents**</DNT> page

>     2. Click on <DNT>**Dashboards**</DNT>

Reviewer: ```suggestion
    2. Click <DNT>**Dashboards**</DNT>
```

    3. In the search bar, type `MSSQL On-Host Integration`

>     4. The MSSQL dashboard should appear. Click on it to install it

Reviewer: ```suggestion
    4. The MSSQL dashboard appears. Click it to install.
```

  </Step>
</Steps>

## Metrics collected by Integration

<CollapserGroup>
  <Collapser
    id="blockingSession"

>     title="Blocking Session Metrics"

Reviewer: ```suggestion
    title="Blocking session metrics"
```

  >
    <table>
      <thead>
      <tr>
        <th>
          Metric
        </th>
        <th>
          Description
        </th>
      </tr>
      </thead>
      <tbody>
      <tr>
        <td>
          `blocking_spid`
        </td>
        <td>
          The ID of the blocking session.
        </td>
      </tr>
      <tr>
        <td>
          `blocking_status`
        </td>
        <td>
          The status of the blocking session.
        </td>
      </tr>
      <tr>
        <td>
          `blocked_spid`
        </td>
        <td>
          The ID of the blocked session.
        </td>
      </tr>
      <tr>
        <td>
          `blocked_status`
        </td>
        <td>
          The status of the blocked session.
        </td>
      </tr>
      <tr>
        <td>
          `wait_type`
        </td>
        <td>
          The type of wait experienced by the blocked session.
        </td>
      </tr>
      <tr>
        <td>
          `wait_time_in_seconds`
        </td>
        <td>
          The time the session has been waiting in seconds.
        </td>
      </tr>
      <tr>
        <td>
          `command_type`
        </td>
        <td>

>           The type of command being executed.

Reviewer: ```suggestion
          The type of command being performed.
```
